src/drt/src/dr/FlexGridGraph.cpp

```diff
@@ -198,20 +213,23 @@ bool FlexGridGraph::hasAlignedUpDefTrack(
   return false;
 }
 
-void FlexGridGraph::initEdges(
-    const frDesign* design,
-    std::map<frCoord, std::map<frLayerNum, frTrackPattern*>>& xMap,
-    std::map<frCoord, std::map<frLayerNum, frTrackPattern*>>& yMap,
-    const std::map<frLayerNum, dbTechLayerDir>& zMap,
-    const Rect& bbox,
-    bool initDR)
+void FlexGridGraph::initEdges(const frDesign* design,
+                              frLayerCoordTrackPatternMap& xMap,
+                              frLayerCoordTrackPatternMap& yMap,
+                              const frLayerDirMap& zMap,
+                              const Rect& bbox,
+                              bool initDR)
 {
   frMIdx xDim, yDim, zDim;
   getDim(xDim, yDim, zDim);
   // initialize grid graph
-  frMIdx xIdx = 0, yIdx = 0, zIdx = 0;
+  frMIdx zIdx = -1;
   dieBox_ = design->getTopBlock()->getDieBox();
   for (const auto& [layerNum, dir] : zMap) {
```

Comment:
is there a reason behind splitting this for loop into 2 for loops (one for each layer direction)?

Reply:
It was an experiment that didn't really pay off, should've rolled it back.